src/util/keyutil.ts

```diff
@@ -0,0 +1,137 @@
+import * as React from "react";
+import * as electron from "electron";
+
+type KeyPressDecl = {
+    mods: Record<ModKeyStrs, boolean>;
+    key: string;
+};
+
+function parseKeyDescription(keyDescription: string): KeyPressDecl {
+    let rtn = { key: "", mods: {} } as KeyPressDecl;
+    rtn.mods.Cmd = false;
+    rtn.mods.Ctrl = false;
+    rtn.mods.Option = false;
+    rtn.mods.Shift = false;
+    let keys = keyDescription.replace(/[()]/g, "").split(":");
+    for (let key of keys) {
+        if (key == "Cmd") {
+            rtn.mods.Cmd = true;
+        } else if (key == "Shift") {
+            rtn.mods.Shift = true;
+        } else if (key == "Ctrl") {
+            rtn.mods.Ctrl = true;
+        } else if (key == "Option") {
+            rtn.mods.Option = true;
+        } else {
+            rtn.key = key;
+            if (key.length == 1) {
+                // check for if key is upper case
+                if (/[A-Z]/.test(key.charAt(0))) {
+                    // this key is an upper case A - Z - we should apply the shift key, even if it wasn't specified
+                    rtn.mods.Shift = true;
+                } else if (key == " ") {
+                    rtn.key = "Space";
+                    // we allow " " and "Space" to be mapped to Space key
+                }
+            }
+        }
+    }
+    return rtn;
+}
+
+function checkKeyPressed(event: WaveKeyboardEvent, description: string): boolean {
+    let keyPress = parseKeyDescription(description);
+    if (keyPress.mods.Option && !event.alt) {
+        return false;
+    }
+    if (keyPress.mods.Cmd && !event.cmd) {
+        return false;
+    }
+    if (keyPress.mods.Shift && !event.shift) {
+        return false;
+    }
+    if (keyPress.mods.Ctrl && !event.control) {
+        return false;
+    }
+    let eventKey = event.key;
+    let descKey = keyPress.key;
+    if (eventKey.length == 1 && /[A-Z]/.test(eventKey.charAt(0))) {
```

Comment:
So this accepts shift:e, shift:E or E all as the same thing
it also accepts "Space" or " " to mean space
Thoughts? 
Also I'm wondering if I should move this logic to the adapt keyboard event functions so that WaveKeyboardEvent has a normalized key value

Reply:
we should add unicode checks in here.  instead of checking for [A-Z] you can use \p{Lu} (unicode uppercase characters).  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape .  I think you also need to set the `v` flag on the regexp.

see also https://unicode.org/reports/tr18/#General_Category_Property